### PR TITLE
Adding capacity limit for MeOH and FT

### PR DESCRIPTION
**Pull request by toniseibold**
Closing https://github.com/PyPSA/pypsa-ariadne/issues/271 and https://github.com/PyPSA/pypsa-ariadne/issues/270
Adding a capacity limit of 2GW and 5GW for Fischer-Tropsch and Methanolisation in Germany respectively.
Furthermore, the production volume limit for hydrogen within Germany is now calibrated towards MWh_H2 and not MWh_electricity.

Before asking for a review for this PR make sure to complete the following checklist:

- [x] Workflow with target rule `ariadne_all` completes without errors
- [ ] The logic of `export_ariadne_variables` has been adapted to the changes
_not applicable_
- [ ] One or several figures that validate the changes in the PR have been posted as a comment
_not applicable_
- [x] A brief description of the changes has been added to `Changelog.md`
- [x] The latest `main` has been merged into the PR
- [x] The config has a new prefix of the format `YYYYMMDDdescriptive_title`

### CODE REVIEW

File: config/config.yaml [hunk 1]
```diff
@@ -485,6 +485,20 @@ solving:
             2035: 101
             2040: 101
             2045: 101
+      Link:
```
Comment (lindnemi) [inline config/config.yaml:488]: This  should be a limit_capacity_max, not _min

File: workflow/scripts/additional_functionality.py [hunk 2]
```diff
@@ -41,16 +41,23 @@ def add_capacity_limits(n, investment_year, limits_capacity, sense="maximum"):
                 extendable_index = c.df.index[
                     valid_components & c.df[attr + "_nom_extendable"]
                 ]
+                efficiency = (
+                    c.df.loc[existing_index, "efficiency"].mean()
```
Comment (lindnemi) [inline workflow/scripts/additional_functionality.py:45]: No need to use `mean`, better multiply the vectors. 

[For FT there should be no problems, but this function is more general and e.g. for gas turbines there would be a difference]

File: workflow/scripts/additional_functionality.py [hunk 3]
```diff
@@ -41,16 +41,23 @@ def add_capacity_limits(n, investment_year, limits_capacity, sense="maximum"):
...
+                    if (c.name == "Link") and (carrier != "electrolysis")
+                    else 1
+                )
 
-                existing_capacity = c.df.loc[existing_index, attr + "_nom"].sum()
+                existing_capacity = (
+                    c.df.loc[existing_index, attr + "_nom"].sum() / efficiency
```
Comment (lindnemi) [inline workflow/scripts/additional_functionality.py:51]: i think the efficiency has to be multiplied, not divided

File: workflow/scripts/additional_functionality.py [hunk 4]
```diff
@@ -41,16 +41,23 @@ def add_capacity_limits(n, investment_year, limits_capacity, sense="maximum"):
...
+                )
 
                 logger.info(
                     f"Existing {c.name} {carrier} capacity in {ct}: {existing_capacity} {units}"
                 )
 
                 nom = n.model[c.name + "-" + attr + "_nom"].loc[extendable_index]
 
-                lhs = nom.sum()
+                lhs = nom.sum() / efficiency
```
Comment (lindnemi) [inline workflow/scripts/additional_functionality.py:60]: again, i think `*` instead of `/`

File: workflow/scripts/additional_functionality.py [hunk 5]
```diff
@@ -231,10 +238,11 @@ def h2_production_limits(n, investment_year, limits_volume_min, limits_volume_ma
         production = n.links[
             (n.links.carrier == "H2 Electrolysis") & (n.links.bus0.str.contains(ct))
         ].index
+        efficiency = n.links.loc[production, "efficiency"].mean()
 
         lhs = (
             n.model["Link-p"].loc[:, production] * n.snapshot_weightings.generators
-        ).sum()
+        ).sum() / efficiency
```
Comment (lindnemi) [inline workflow/scripts/additional_functionality.py:245]: lines 241 and 245, same as above

**Review by lindnemi — CHANGES_REQUESTED** [review]

**Comment by toniseibold** [comment]
Bug: Constraint active (e.g. in 2035 mu != 0) but capacities not built to the maximum

**Comment by JulianGeis** [comment]
the constraint seem to be fine:
image

However, there is something going wrong in the optimisation. TBC

**Comment by lindnemi** [comment]
superseded by #278 and some direct commits to main

**Pull request closed without merging**